service/src/main/java/io/camunda/service/UserTaskServices.java

```diff
@@ -123,17 +164,17 @@ public CompletableFuture<UserTaskRecord> updateUserTask(
   }
 
   public UserTaskEntity getByKey(final long userTaskKey) {
+    final var query = UserTaskQuery.of(q -> q.filter(f -> f.userTaskKeys(userTaskKey)));
     final var result =
-        userTaskSearchClient
-            .withSecurityContext(securityContextProvider.provideSecurityContext(authentication))
-            .searchUserTasks(userTaskSearchQuery(q -> q.filter(f -> f.userTaskKeys(userTaskKey))));
-    final var userTaskEntity = getSingleResultOrThrow(result, userTaskKey, "User task");
+        search(query, securityContextProvider.provideSecurityContext(authentication));
```

Comment:
@tmetzke this is the part we might wanna sync on.

Right now, the tests expects a `Forbidden` error, but when I use the search here with default authorizations, then it become a `NotFound` - which makes sense to me 😅 (because the user can not read this entity)

I would in general be in favor of returning `NotFound` when someone calls `getByKey` and not having the permissions to read the entity. As the fact that we return `Forbidden` is an information the caller of this endpoint can abuse -> for example, as it says "Hey we have this USER, but you can not access it" allows an attacker to collect a bunch of known users just by bruteforcing such an endpoint.

Reply:
100% agree with your reasoning, we do the same for missing tenant access 👍